core-bundle/contao/drivers/DC_Table.php

```diff
@@ -4411,7 +4411,8 @@ protected function generateTree($table, $id, $arrPrevNext, $blnHasSorting, $intM
 
 		// Calculate label and add a toggle button
 		$level = $intMargin / $intSpacing + 1;
-		$blnIsOpen = isset($session[$node][$id]) && $session[$node][$id] == 1;
+		$found = \count($arrFound);
+		$blnIsOpen = ($found > 0 && $found < 50) || ($session[$node][$id] ?? null) == 1;
```

Comment:
```suggestion
		$blnIsOpen = !empty($arrFound) || ($session[$node][$id] ?? null) == 1;
```

After using the fix for some time in a big installation, I agree that it is better if we always open the nodes if there are search results.